### PR TITLE
Get visitorId from localstorage

### PR DESCRIPTION
**Pull request by erocheleau**
Get visitorId from localstorage

Added scratch org def settings

### CODE REVIEW

**Review by ebouchard2 — APPROVED** [review]

File: src/main/default/lwc/caseAssistFlow/caseAssistFlow.js [hunk 1]
```diff
@@ -40,10 +40,15 @@ export default class CaseAssistFlow extends LightningElement {
     // Debounce function for "search-as-you-type" it will trigger when there is no key stroke for 500ms.
     this.debounceHandler = debounce(async () => {
       try {
+        // If you have the AnalyticsBeacon LWC in your community, it will have generated a visitorId for you by now.
+        const visitorId = localStorage.getItem('visitorId');
```
Comment (lbergeron) [inline src/main/default/lwc/caseAssistFlow/caseAssistFlow.js:44]: Can you add tests to validate that `visitorId` is read from the storage?

**Review by lbergeron — CHANGES_REQUESTED** [review]
Tests should be added to validate that `visitorId` is read from the storage.

File: src/main/default/lwc/caseAssistFlow/caseAssistFlow.js [hunk 2]
```diff
@@ -40,10 +40,15 @@ export default class CaseAssistFlow extends LightningElement {
     // Debounce function for "search-as-you-type" it will trigger when there is no key stroke for 500ms.
     this.debounceHandler = debounce(async () => {
       try {
+        // If you have the AnalyticsBeacon LWC in your community, it will have generated a visitorId for you by now.
```
Comment (louis-bompart) [inline src/main/default/lwc/caseAssistFlow/caseAssistFlow.js:43]: I think it'd make sense for the AnalyticsBeacon to have a method that uses the Storage class of CA.js
You can get the storage instance using `window.coveoanalytics.storage.getAvailableStorage()`

You could wrap this in a utility class (not a component) and then refer to this class and its function:
image

**Review by louis-bompart — COMMENTED** [review]
I think it'd be nice to change how we get the visitorId

**Review by louis-bompart — APPROVED** [review]

**Review by lbergeron — APPROVED** [review]

**Pull request merged**